[tox]
envlist = py35,py36,py27,lint

[testenv]
# Some versions of Pytest versions have a bug:
# https://github.com/pytest-dev/pytest/issues/5971 which causes the whole test
# process to crash if a multiprocessing job has an exception. Fixed in 5.3.3.
deps =
    -r{toxinidir}/requirements.txt
    pytest>=5.4.1
    pytest-xdist==1.32.0
    filelock
commands =
    py.test -v --basetemp={envtmpdir} --junit-xml={toxinidir}/python_tests.xml --junit-prefix={envname} --color=yes -n 1 {posargs}

[testenv:lint]
basepython = python3
deps =
    -r{toxinidir}/requirements.txt
    flake8
commands = flake8 {toxinidir}/Tensile

[testenv:pre_commit]
basepython = python3
deps =
    -r{toxinidir}/requirements.txt
    pytest>=5.4.1
    flake8
commands =
    flake8 {toxinidir}/Tensile
    py.test -q --basetemp={envtmpdir} --color=yes {toxinidir}/Tensile -m unit {posargs}

[flake8]
exclude = .git,build*,dist,.cache,*~
max-line-length = 132
ignore =
     # All E class violations are errors reported by pycodestyle
     E,
     # All W class violations are warnings reported by pycodestyle
     W
     # This mostly just leaves pyflakes errors.
